commit 97368cfb97e4adb3b92e42db9e431d2ee94eb643
Author: Leann Ogasawara <leann.ogasawara@canonical.com>
Date:   Fri Sep 9 13:23:51 2011 -0700

    UBUNTU: SAUCE: (no-up) x86: reboot: Make Dell Optiplex 790 use reboot=pci
    
    BugLink: http://bugs.launchpad.net/bugs/818933
    
    The Dell Optiplex 790 doesn't reboot unless reboot=pci is set.
    
    Signed-off-by: Leann Ogasawara <leann.ogasawara@canonical.com>
    Signed-off-by: Tim Gardner <tim.gardner@canonical.com>

diff --git a/arch/x86/kernel/reboot.c b/arch/x86/kernel/reboot.c
index acc362a26fb8..3959c2e15e28 100644
--- a/arch/x86/kernel/reboot.c
+++ b/arch/x86/kernel/reboot.c
@@ -465,6 +465,14 @@ static const struct dmi_system_id reboot_dmi_table[] __initconst = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "Latitude E6520"),
 		},
 	},
+	{       /* Handle problems with rebooting on the OptiPlex 790. */
+		.callback = set_pci_reboot,
+		.ident = "Dell OptiPlex 790",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Dell Inc."),
+			DMI_MATCH(DMI_PRODUCT_NAME, "OptiPlex 790"),
+		},
+	},
 	{ }
 };
 
